pkg/monitoring/alloy/configmap.go

```diff
@@ -82,11 +85,16 @@ func (a *Service) GenerateAlloyMonitoringConfigMapData(ctx context.Context, curr
 		PriorityClassName string
 		Replicas          int
 		SecretName        string
+
+		SupportVPA bool
 	}{
 		AlloyConfig:       alloyConfig,
 		PriorityClassName: commonmonitoring.PriorityClassName,
 		Replicas:          shards,
 		SecretName:        commonmonitoring.AlloyMonitoringAgentAppName,
+
+		// Observability bundle in older versions do not support VPA
+		SupportVPA: observabilityBundleVersion.GE(supportVPA),
```

Comment:
I'm not comfortable with the naming here:
- supportVPA is a semver
- SupportVPA is a boolean
I think it's quite tricky and could be a source of confusion.

Reply:
Does IsSupportingVPA good for you?

Reply:
Done :)